src/html/context.rs

```diff
@@ -0,0 +1,88 @@
+use crate::html::{self, Anchorizer};
+use crate::{Options, Plugins};
+
+use std::cell::Cell;
+use std::io::{self, Write};
+
+/// Context struct given to formatter functions as taken by
+/// [`html::format_document_with_formatter`].  Output can be appended to through
+/// this struct's [`Write`] interface.
+pub struct Context<'o, 'c> {
+    output: &'o mut dyn Write,
+    last_was_lf: Cell<bool>,
+
+    /// [`Options`] in use in this render.
+    pub options: &'o Options<'c>,
+    /// [`Plugins`] in use in this render.
+    pub plugins: &'o Plugins<'o>,
+
+    pub(super) anchorizer: Anchorizer,
```

Comment:
I think we should make this public in general. I'm trying to override `render_heading` and would need access to the `anchorizer`